src/symbol.rs

```diff
@@ -6,34 +6,6 @@ use crate::parse::{ParseAt, ParseError, ParsingTable};
 
 pub type SymbolTable<'data, E> = ParsingTable<'data, E, Symbol>;
 
-/// C-style 32-bit ELF Symbol definition
-///
-/// These C-style definitions are for users who want to implement their own ELF manipulation logic.
-#[derive(Debug)]
-#[repr(C)]
-pub struct Elf32_Sym {
-    pub st_name: u32,
-    pub st_value: u32,
-    pub st_size: u32,
-    pub st_info: u8,
-    pub st_other: u8,
-    pub st_shndx: u32,
```

Comment:
`st_shndx` type should be `uint16_t`.

reference:
1. https://llvm.org/doxygen/structllvm_1_1ELF_1_1Elf32__Sym.html
2. https://man7.org/linux/man-pages/man5/elf.5.html